php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -2300,29 +2300,33 @@ abstract class NDB_BVL_Instrument extends NDB_Page
         $config   = NDB_Config::singleton();
         $db       = Database::singleton();
         $dbconfig = $config->getSetting('database');
-        $columns  = $db->pselect(
-            "SELECT COLUMN_NAME FROM information_schema.columns
+        if ($this->jsonData) {
+            $db->update('flag', ['Data'=>null], ['CommentID' => $this->commentID]);
```

Comment:
Shouldn't this update be unconditional? The `Data` column in the flag table exists regardless of whether jsonData is true or not, and the update when saving data in `_save` isn't conditional.

Reply:
You are correct. Ma bad